lib/pubid/iso/identifier/base.rb

```diff
@@ -134,6 +134,8 @@ def self.parse_from_title(title)
           # delete parts from the title until it's parseable
           acc.reverse.sub(part.reverse, "").reverse.strip
         end
+
+        raise Errors::ParseError, "cannot parse #{title}"
```

Comment:
Can we have more specific errors such as “InvalidPrefixError” (for disallowed prefixes), “InvalidTypedStageError” etc? Thanks.

Reply:
> Can we have more specific errors such as “InvalidPrefixError” (for disallowed prefixes), “InvalidTypedStageError” etc? Thanks.

Parsing error is a result of Parslet's parsing error `Parslet::ParseFailed`, there are no way to extract the cause of error, only by reading long error message.
To get specific error, we need to change parsing and parse it step by step, could be quite challenging tasks because of many formats.

Reply:
Let's merge this first and then deal with more fine grained errors. Thanks.